File: train.cpp
//------------------------------------------------------------------------------
// train.cpp - содержит функции обработки самолета
//------------------------------------------------------------------------------

#include "train.h"
#include <math.h>

//------------------------------------------------------------------------------
// Подсчет рандомного действительного значения
double randomT(double b, double e)
{
    return e + fabs(Random() - b) * 100;
}

//------------------------------------------------------------------------------
// Ввод параметров поезда из потока
void In(train &t, std::ifstream &ifst) {
    ifst >> t.number_wagons >> t.speed >> t.range;
}

// Случайный ввод параметров поезда
void InRnd(train &t) {
    t.range = randomT(1.54, 100);
    t.speed = Random();
    t.number_wagons = Random();
}

//------------------------------------------------------------------------------
// Вывод параметров поезда в поток
void Out(train &t, std::ofstream &ofst) {
    ofst << "Train specifications: number of wagons = "
         << t.number_wagons
         << ", Speed = " << t.speed
         << ", Range = " << t.range
         << ". Ideal time of passage = " << CountTime(t) << "\n";
}


//------------------------------------------------------------------------------
// Идеальное время прохождения пути для поезда

double CountTime(train &t) {
    double time = t.range / (double)t.speed;
    return time;
}

File: HW3/src/train.py
import newRandom


# ----------------------------------------------
# Класс поезда
class Train:
    # Конструктор поезда
    def __init__(self):
        self.range = 0
        self.speed = 0
        self.number_wagons = 0

    # Считывание параметров поезда из строки
    def ReadStrArray(self, strArray, i):
        if i >= len(strArray) - 1:
            return 0
        self.number_wagons = int(strArray[i])
        self.speed = int(strArray[i + 1])
        self.range = int(strArray[i + 2])
        i += 3
        return i

    # Случайная генерация параметров поезда
    def RandomIn(self):
        self.range = newRandom.randDouble(10, 10000)
        self.speed = newRandom.randInt(100, 10000)
        self.number_wagons = newRandom.randInt(1, 50)

    # Вывод данных поезда в файл
    def Write(self, ostream):
        ostream.write("Train: number of wagons = {}, speed = {}, range = {}, Ideal time of travel = {}"
                      .format(self.number_wagons, self.speed, self.range, self.CountTime()))
        pass

    #  Идеальное время прохождения пути для поезда
    def CountTime(self):
        return self.range / self.speed
